### PR TITLE
Performance: Stop using pkg_resources/declare_namespace 

### PR DESCRIPTION
**Pull request by johanste**
Loading pkg_resources and calling declare_namespace caused startup performance to suffer. This change adopts the solution from the azure sdk, where pkg_resources are only used for dev/editable installs and are stripped out when building wheels. 

In order to be able to reuse the azure_bdist_wheel.py wheel extension implementation from the azure sdk, we also introduced an additional namespace package (the name of the namespace package is used to determine which __init__.py files to exclude when building the wheel)
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [N/A] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [N/A] Each command and parameter has a meaningful description.
- [N/A] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-command_modules-nspkg/setup.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    description='Microsoft Azure CLI Command Modules Namespace Package',
+    long_description=README + '\n\n' + HISTORY,
+    license='MIT',
+    author='Microsoft Corporation',
+    author_email='azpycli@microsoft.com',
+    url='https://github.com/Azure/azure-cli',
+    zip_safe=False,
+    classifiers=CLASSIFIERS,
+    install_requires=[
+        'azure-cli-nspkg>=1.0.0'
```
Comment (lmazuel) [inline src/azure-cli-command_modules-nspkg/setup.py]: should be 3.0.0

File: src/azure-cli-core/setup.py [hunk 2]
```diff
@@ -86,16 +92,15 @@
...
+        'azure',
+        'azure.cli',
         'azure.cli.core',
         'azure.cli.core.commands',
         'azure.cli.core.extensions',
         'azure.cli.core.sdk',
         'azure.cli.core.profiles',
     ],
-    install_requires=DEPENDENCIES
+    install_requires=DEPENDENCIES,
```
Comment (lmazuel) [inline src/azure-cli-core/setup.py]: I can't comment upper, but if you want setuptools to work out of the box, you should remove "azure-cli-nspkg" as a direct dependencies of the package. This way the "setuptools" install does not install it, and azure_bdist_wheel will add it automatically to the wheel.

Comment (johanste) [inline src/azure-cli-core/setup.py]: Yep - thanks, missed that. Fixed!

File: src/azure-cli-command_modules-nspkg/setup.py (hunk 1 above)
Comment (johanste) [inline src/azure-cli-command_modules-nspkg/setup.py]: Yep. Done.

**Comment by lmazuel** [comment]
@johanste LGTM but didn't clone the branch. If you want I can clone it tomorrow and build to check everything is fine

**Comment by johanste** [comment]
@lmazuel, we would very much appreciate you cloning & building. We do intend to merge today.

**Pull request merged**